src/proof/dot/dot_printer.cpp

```diff
@@ -235,13 +235,25 @@ uint64_t DotPrinter::printInternal(
     ProofNodeHashFunction hasher;
     size_t currentHash = hasher(pn);
 
+    // Search if the current hash is in the vector
+    std::reverse_iterator<std::vector<size_t>::iterator> it;
+    for (it = ancestorHashs.rbegin(); it != ancestorHashs.rend(); ++it)
+    {
+      if (*it == currentHash)
+      {
+        break;
+      }
+    }
```

Comment:
I think the probability to find an equal hash is bigger in the end of the vector, since to have the same hash both nodes have to share close ancestors/descendants. Therefore, I think the reverse traverse fits better.